File: Java/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla {
    private MazoSorpresas mazo;

    public CasillaSorpresa(MazoSorpresas _mazo, String nombre) {
        super(nombre);
        mazo = _mazo;
    }

    @Override
    public String toString() {
        return super.toString() + "\nLa sorpresa que saldrá es:\n" + sorpresa.toString() + "\n";
    }

    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (super.jugadorCorrecto(actual, todos)) {
            super.informe(actual, todos);
            sorpresa = mazo.siguiente();
            this.informe(actual, todos);
            sorpresa.aplicarAJugador(actual, todos);
        }
    }
}

File: Ruby/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaSorpresa
    def initialize (mazo,nombre)
      @mazo = mazo
      super(nombre)
    end
    
    def recibe_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        @sorpresa = @mazo.siguiente() 
        super
        @sorpresa.aplicar_a_jugador(actual, todos)
      end
    end
    
    def to_s
      return super + "\nLa sorpresa que saldrá es:\n" + @sorpresa.to_s() + "\n"
    end
  end
end
